src/OpenSage.Game/Logic/Object/Draw/W3dModelDraw.cs

```diff
@@ -261,6 +263,26 @@ public override void UpdateConditionState(BitArray<ModelConditionFlag> flags, Ra
             SetActiveAnimationState(bestAnimationState, random);
         }
 
+        private void UpdateBoneVisibilities(ModelConditionState conditionState, W3dModelDrawConditionState drawState)
+        {
+            var model = conditionState.Model?.Value;
+            if (model == null) return;
+
+            foreach (var subObject in model.SubObjects)
+            {
+                var name = subObject.Name;
+
+                if ((subObject.RenderObject.Hidden && !(Drawable.ShownSubObjects?.ContainsKey(name) ?? false))
+                    || (Drawable.HiddenSubObjects?.ContainsKey(name) ?? false))
```

Comment:
These should never be null as you've made them autoproperties for the readonly properties beneath them, and those are assigned a new dictionary on object construction
```suggestion
                if ((subObject.RenderObject.Hidden && !Drawable.ShownSubObjects.ContainsKey(name))
                    || Drawable.HiddenSubObjects?.ContainsKey(name))
```

Reply:
Looking at this a little closer - does this need some extra parentheses? && takes conditional precedence https://stackoverflow.com/questions/1196703/c-sharp-conditional-and-or-precedence

Reply:
i may be missing something but doesn't it already have the extra parentheses?
```csharp
(subObject.RenderObject.Hidden && !Drawable.ShownSubObjects.ContainsKey(name))
                    || Drawable.HiddenSubObjects?.ContainsKey(name)
```
boils down to
```
(true && false) || true
```
or if you meant that it DOESN'T need them, well, i think it's just more obvious what's happening when you're reading that with parentheses

Reply:
removed the null checks

Reply:
> i may be missing something but doesn't it already have the extra parentheses?
> 
> ```cs
> (subObject.RenderObject.Hidden && !Drawable.ShownSubObjects.ContainsKey(name))
>                     || Drawable.HiddenSubObjects?.ContainsKey(name)
> ```
> 
> boils down to
> 
> ```
> (true && false) || true
> ```
> 
> or if you meant that it DOESN'T need them, well, i think it's just more obvious what's happening when you're reading that with parentheses

Honestly I have no idea where they _should_ be. I was assuming `a && (b || c)` was desired, but I don't know 🤷

Reply:
well I think the desired logic is this:
if the object is marked as hidden in the model, and is not flagged as forcefully shown by the Drawable, then hide it. also if the object is forcefully hidden by the Drawable, then hide it too

Reply:
adding to that, we shouldn't have cases where a subobject is both marked hidden and shown in the drawable

Reply:
Never say never 😆